### PR TITLE
Investigate and provide some API documentation tool

### PR DESCRIPTION
**Pull request by dsounded**
https://app.asana.com/0/156934457071997/156934457072000

### CODE REVIEW

**Comment by dsounded** [comment]
Even if `travis` fails I'd like to fix specs in separate request, `cause this is previous issue, anyway probably I already know how to fix it

File: app/controllers/CountriesController.scala [hunk 1]
```diff
@@ -13,8 +13,15 @@ import models.{Countries,Country}
...
 import responders.Responder
 
+import io.swagger.annotations._
+
+@Api(value = "/countries", description = "Countries CRUD", consumes="application/json")
 @Singleton
 class CountriesController @Inject() extends Controller {
+  @ApiOperation(httpMethod = "GET", value = "Get all countries")
+  @ApiResponses(Array(
+    new ApiResponse(code = 200, message = "{'countries':[{'id':1,'title':'Ukraine','abbreviation':'UA','isActive':true,'createdAt':1468860114000}]}")
```
Comment (hydro0) [inline app/controllers/CountriesController.scala]: http://docs.swagger.io/swagger-core/v1.3.12/apidocs/index.html?com/wordnik/swagger/annotations/ApiResponses.html

As from swagger documentation, message should be human-readable description about what means this response, and model should be passed as response field - classOf[Array[Country]]

Comment (dsounded) [inline app/controllers/CountriesController.scala]: Yeah, but shouldn't it be Json[Array[Country]] ? Plus, where you suggest to put response example ?

Comment (hydro0) [inline app/controllers/CountriesController.scala]: new ApiResponse(code = 200, message = "Country was successfully retrieved", response = classOf[Array[Country]])
I mean smthing like that. Swagger automatically generates examples of model and its JSON representation.

Comment (dsounded) [inline app/controllers/CountriesController.scala]: Hmm, l've seen the response of this lib and found no such am example :)

File: app/controllers/UsersController.scala [hunk 2]
```diff
@@ -36,6 +47,21 @@ class UsersController @Inject() extends Controller {
...
 
+  @ApiOperation(httpMethod = "POST", value = "Create user", response = classOf[User], responseContainer = "Json.obj")
+  @ApiResponses(Array(
+    new ApiResponse(code = 201, message = "Created"),
+    new ApiResponse(code = 404, message = "Bad Request")
+    ))
+  @ApiImplicitParams(Array(
+    new ApiImplicitParam(name = "user[email]", value = "User's email", required = true, dataType = "String", paramType = "body"),
+    new ApiImplicitParam(name = "user[password]", value = "User's password", required = true, dataType = "String", paramType = "body"),
+    new ApiImplicitParam(name = "country[firstName]", value = "User's first name", required = false, dataType = "String", paramType = "body"),
```
Comment (hydro0) [inline app/controllers/UsersController.scala]: country[firstName]  :)

**Comment by dsounded** [comment]
@hydro0 I'd like to merge this and consider what should we do with `root` request keys later

**Comment by hydro0** [comment]
Ship it!

**Pull request merged**